### PR TITLE
docs: add Ada to the dependencies page

### PR DESCRIPTION
**Pull request by anonrig**
The current dependencies list is not up to date. The following changes only updates `Ada`, Node.js's new URL parser.

### CODE REVIEW

**Review by UlisesGascon — APPROVED** [review]

**Review by tniessen — COMMENTED** [review]
This list is _very_ incomplete. Higher-impact dependencies such as undici and nghttp2 are not listed either. I am not sure what subset of dependencies makes sense here, or if this page provides value at all in its current state.

**Comment by ovflowd** [comment]
@tniessen the page is just super outdated. But would love if it can be updated with all the new stuff :D

**Comment by Trott** [comment]
I'm not sure this page adds value for the average user. I wonder if it should be ported to a README for the `deps` directory in Node.js source. Then we can replace the page with a link there. That is far more likely to be updated when people add/remove dependencies. And because the README is associated with a particular commit, it will reflect the status at that commit. In other words, we avoid the problem of trying to write a single page that reflects all versions of Node.js.

**Comment by Trott** [comment]
> I'm not sure this page adds value for the average user. I wonder if it should be ported to a README for the `deps` directory in Node.js source. Then we can replace the page with a link there. That is far more likely to be updated when people add/remove dependencies. And because the README is associated with a particular commit, it will reflect the status at that commit. In other words, we avoid the problem of trying to write a single page that reflects all versions of Node.js.

None of this is an objection to updating the page to reflect current-ish information though.

File: pages/en/docs/meta/topics/dependencies.md
Comment (Trott) [inline pages/en/docs/meta/topics/dependencies.md]:
```suggestion
For fast URL parsing, Node.js relies on a [WHATWG Spec](https://url.spec.whatwg.org)
```

File: pages/en/docs/meta/topics/dependencies.md
Comment (Trott) [inline pages/en/docs/meta/topics/dependencies.md]:
```suggestion
module resolution, network requests, and filesystem.
```

**Comment by ovflowd** [comment]
That's correct, @Trott. We can close this PR and pivot to doing something on `node/deps/README.md`. I mention this because I remembered the nodejs.dev "about" doesn't mention any of these dependencies.

Meaning even if we add this now, as we agreed that the .dev pages replace the .org ones, it will be removed very prematurely.

@anonrig do you think you can pivot such kind of change on `@nodejs/node/deps/README.md`?

**Comment by anonrig** [comment]
> And because the README is associated with a particular commit, it will reflect the status at that commit.

@Trott Even though, I agree with your concerns, this particular recommendation will lead to outdated commit URL in this page too. If we are going this path, I think we should include the dependencies and distribute it with the Node.js documentation (which is version specific). (I didn't had any chance to look into the documentation tools & infrastructure. I don't know the depth or possibility of this task, therefore my assumption/recommendation might be invalid.)

> None of this is an objection to updating the page to reflect current-ish information though.

@Trott I recommend merging this pull request, and later discuss the future of this document in a separate PR or an issue for wider visibility.

> Meaning even if we add this now, as we agreed that the .dev pages replace the .org ones, it will be removed very prematurely.

@ovflowd If this is the case, we should remove those documents from Crowdin. A lot of people are investing their own time to translate these soon-to-be-removed documents, and if we are simply not use them in the future, we should suspend all translation activities.

> @anonrig do you think you can pivot such kind of change on @nodejs/node/deps/README.md?

@ovflowd As a non-native English speaker, I think there are more qualified people who can do this faster & cleaner than me.

**Comment by ovflowd** [comment]
@anonrig Ive made an announcement on Crowdin several days ago asking people to no translate the current content as it's going to be 100% discarded 🙈

**Review by amantank — COMMENTED** [review]
Outdated.

**Comment by AugustinMauroy** [comment]
@amantank can you give more detail ?

**Comment by bmuenzenmeyer** [comment]
> @ovflowd As a non-native English speaker, I think there are more qualified people who can do this faster & cleaner than me.

I'd be happy to draft this up!

**Comment by bmuenzenmeyer** [comment]
I was about to propose a new README at [`deps/README.md`](https://github.com/nodejs/node/tree/main/deps), but a quick scan of the current codebase found an existing and probably better source: [doc/contributing/maintaining/maintaining-dependencies](https://github.com/nodejs/node/blob/main/doc/contributing/maintaining/maintaining-dependencies.md#dependency-list)

Taking @Trott's words here as a guiding set of principles, this file seems to have everything we would want:

- `I'm not sure this page adds value for the average user. ` - this is a contributing file. Agree that most users won't even know or care about internals
- ` I wonder if it should be ported to a README for the 'deps' directory in Node.js source.` - while perhaps it might be nice to duplicate a list or link to the `maintaining-dependencies`, I'd rather defer to the project maintainers over in nodejs/node that have not yet found reason to create a new README. This is not the case for a directory like https://github.com/nodejs/node/tree/main/test which does have useful and unique info
- `Then we can replace the page with a link there.` - would this get too into the weeds of maintaining direct from the website? I'd argue no, as we are already discussing linking directly into GitHub.
- `That is far more likely to be updated when people add/remove dependencies. And because the README is associated with a particular commit, it will reflect the status at that commit. In other words, we avoid the problem of trying to write a single page that reflects all versions of Node.js.` - the organic use of  [doc/contributing/maintaining/maintaining-dependencies](https://github.com/nodejs/node/blob/main/doc/contributing/maintaining/maintaining-dependencies.md#dependency-list) already qualifies, citing a recent update like https://github.com/nodejs/node/pull/48618 that maintained this file

---

`tl;dr`: merge this as discused previously, but eventually shift toward linking directly to https://github.com/nodejs/node/blob/main/doc/contributing/maintaining/maintaining-dependencies.md#dependency-list or https://github.com/nodejs/node/blob/main/doc/contributing/maintaining/maintaining-dependencies.md

**Comment by ovflowd** [comment]
I think we should delete this page and make the link on navigation to simply redirect to the Markdown file on core.

**Pull request closed without merging**
